src/js/components/purchaseOrder/PurchaseOrderListTable.jsx

```diff
@@ -125,17 +123,15 @@ const PurchaseOrderListTable = ({
     });
   };
 
-  const rollbackOrder = (id) => {
-    apiClient.post(`/openboxes/api/purchaseOrders/${id}/rollback`)
-      .then((response) => {
-        if (response.status === 200) {
-          Alert.success(translate(
-            'react.purchaseOrder.rollback.success.label',
-            'Rollback of order status has been done successfully',
-          ));
-          fireFetchData();
-        }
-      });
+  const rollbackOrder = async (id) => {
+    const { status } = await purchaseOrderApi.rollbackOrder(id);
+    if (status === 200) {
+      Alert.success(translate(
+        'react.purchaseOrder.rollback.success.label',
+        'Rollback of order status has been done successfully',
+      ));
+      fireFetchData();
+    }
```

Comment:
I know that you are not the author, but I believe there also should be spinner added. Would you mind adding it?